packages/entity/src/EntityFields.ts

```diff
@@ -1,16 +1,98 @@
+import { IEntityClass } from './Entity';
+import EntityPrivacyPolicy from './EntityPrivacyPolicy';
+import ReadonlyEntity from './ReadonlyEntity';
+import ViewerContext from './ViewerContext';
+
+export enum EntityEdgeDeletionBehavior {
+  /**
+   * Default. Invalidate the cache for all entities that reference the entity
+   * being deleted through this field. This is most useful when the database itself expresses foreign
+   * keys and cascading deletes or set nulls and the entity framework just needs to
+   * be kept consistent with the state of the database.
+   */
+  INVALIDATE_CACHE,
+
+  /**
+   * Delete all entities that reference the entity being deleted through this field. This is very similar
+   * to SQL `ON DELETE CASCADE` but is done in the Entity framework instead of at the underlying level,
+   * and should not be used in combination with database-schema-expressed foreign keys and deletion behavior.
+   */
+  CASCADE_DELETE,
+
+  /**
+   * Set this field to null when the referenced entity is deleted. This is very similar
+   * to SQL `ON DELETE SET NULL` but is done in the Entity framework instead of at the underlying level,
+   * and should not be used in combination with database-schema-expressed foreign keys and deletion behavior.
+   */
+  SET_NULL,
+}
```

Comment:
It's worth adding that `CASCADE_DELETE` and `SET_NULL` also have the default `INVALIDATE_CACHE` behavior in the comments